larq/callbacks.py

```diff
@@ -52,20 +64,32 @@ def set_model(self, model: keras.models.Model):
                 f'Optimizer must have a "{self.hyperparameter}" attribute.'
             )
 
-    def on_epoch_begin(self, epoch: int, logs: Optional[Dict] = None):
+    def set_hyperparameter(self, t: int):
         hp = getattr(self.optimizer, self.hyperparameter)
         try:  # new API
             hyperparameter_val = tf.keras.backend.get_value(hp)
-            hyperparameter_val = self.schedule(epoch, hyperparameter_val)
+            hyperparameter_val = self.schedule(t, hyperparameter_val)
         except TypeError:  # Support for old API for backward compatibility
-            hyperparameter_val = self.schedule(epoch)
+            hyperparameter_val = self.schedule(t)
 
         tf.keras.backend.set_value(hp, hyperparameter_val)
+        return hp
 
-        if self.verbose > 0:
-            print(
-                f"Epoch {epoch + 1}: {self.hyperparameter} changing to {tf.keras.backend.get_value(hp)}."
-            )
+    def on_epoch_begin(self, epoch: int, logs: Optional[Dict] = None):
+        if self.unit == "epoch":
+            hp = self.set_hyperparameter(epoch)
+
+            if self.verbose > 0:
+                print(
+                    f"Epoch {epoch + 1}: {self.hyperparameter} changing "
+                    + f"to {tf.keras.backend.get_value(hp)}."
+                )
+
+    def on_batch_begin(self, batch: int, logs: Optional[Dict] = None):
+        if self.unit == "step":
+            # We use optimizer.iterations (i.e. global step), since batch only
+            # reflects the batch index in the current epoch.
+            self.set_hyperparameter(self.optimizer.iterations)
```

Comment:
Maybe add an `if verbose: print(...)` here too, since it's also there for the epoch version.